EIPS/eip-7917.md

```diff
@@ -0,0 +1,103 @@
+
+## Backwards Compatibility
+
+No backward compatibility issues found.
+
+## Test Cases
+
+See [the spec tests](https://github.com/ethereum/consensus-specs/tree/6a1d923730c0b368b5192e64f643f22c21451058/tests/core/pyspec/eth2spec/test/fulu) in consensus-specs.
+
+## Security Considerations
+
+### First Block After the Fork
+
+The first block after the fork will calculate all lookaheads for epochs up to `MIN_SEED_LOOKAHEAD` ahead and fill the `proposer_lookahead` field in the beacon state.
+
+### Malicious Lookahead Alteration
+
+A critical consideration for any proposer election mechanism is preventing validators from manipulating lookahead to gain an unfair advantage. This proposal does not increase the chances of such attacks, as we do not alter the “RANDAO delay” used in the lookahead—the lookahead of epoch `N` is still determined by the RANDAO of epoch `N - MIN_SEED_LOOKAHEAD - 1` (which becomes available at the start of epoch `N - MIN_SEED_LOOKAHEAD`). The only difference is that it changes the “effective balances delay”: rather than using the effective balances (EB) at the start of epoch `N`, it now uses the EB at the start of epoch `N - MIN_SEED_LOOKAHEAD`.
+
+Furthermore, by aligning the RANDAO and effective balances in this way, the proposal removes any chance of validators adjusting their EB after seeing the RANDAO outcome, which is an attack vector to consider. No such attack has been found so far, but this change removes the possibility, hence simplifying the security analysis.
+
+### Single Secret Leader Election Compatibility
+
+In the future, we may introduce a Single Secret Leader Election (SSLE) mechanism in which only the selected validator knows their role until they propose a block. However, current SSLE designs still rely on a lookahead, albeit an encrypted one. In such designs, we could reuse the `proposer_lookahead` field by changing its type to something like `List[EncryptedValidatorIndex]`. And if a construction were to remove lookahead entirely, we could simply set `proposer_lookahead` to an empty list, meaning this wouldn’t be a blocker.
+
+That said, any such changes would introduce additional complexity around preconfirmation protocols, but that complexity arises regardless of this EIP.
+
+### Added Computation at the Epoch Boundary
+
+Before this EIP, consensus clients only needed to compute the current proposer’s index at each slot. With the changes introduced by this EIP, they must calculate the entire epoch’s proposer schedule at the start of each epoch. However, computing the proposer index is light, involving sampling validators until we reach a validator with sufficient effective balance to be selected as a proposer. Still, testing is needed to confirm that these additional calculations do not create performance bottlenecks in practice.
```

Comment:
again not a security consideration, move to specification

Reply:
Agreed again, moved to the "rational" part too https://github.com/ethereum/EIPs/pull/9530/commits/e9232b8bf1c74e4b516c809d6d54b66246fe17c3

Thanks!